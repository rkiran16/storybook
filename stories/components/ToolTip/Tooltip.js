import { html } from "lit-html";

const ToolTipContent = () => `
    <div class="text-start">
      <p class="mb-1">
          This is a few lines of tooltip text. 
          This box shows the maximum amount of content that should be displayed inside.
          If more room is needed, use a modal instead.
      </p>   
      <a type="button" href="#" class="btn btn-tertiary">Tertiary Button</a> 
    </div>  
`

export const ToolTip = ({showText = false}) => {
   return html`
    <script>
      window.addEventListener("DOMContentLoaded", () => {
        var tooltipTriggerList = [].slice.call(
          document.querySelectorAll('[data-bs-toggle="tooltip"]')
        );
        var tooltipList = tooltipTriggerList.map(function (tooltipTriggerEl) {
          return new bootstrap.Tooltip(tooltipTriggerEl);
        });
      });
    </script>
    <button class="btn btn-link link-secondary text-decoration-none d-flex align-items-center p-2" data-bs-toggle="tooltip" data-bs-html="true" data-bs-placement="bottom" title=${ToolTipContent()}>
      ${showText ? html`<span class="text-decoration-underline me-2">Tooltip Text</span>` : ''}
      <span class="material-icons-outlined m-0">info</span>
    </button>
   `
}