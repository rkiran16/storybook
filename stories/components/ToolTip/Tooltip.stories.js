import { ToolTip } from "./Tooltip";

export default {
  title: 'Components/ToolTip',
  parameters: {
    zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/630d28ac8bfb866494ca7a35",
  },
  argTypes: {
    showText: {
      type: { name: 'boolean', required: false},
      description: "Show Text",
      defaultValue: false,
      table: {
        type: { summary: 'boolean' },
        defaultValue: { summary: false },
      },
      control: {
        type: 'boolean'
      }
    }
  }
}

const Template = args => ToolTip(args);

export const Icon_Only = Template.bind({});

export const With_Text = Template.bind({});
With_Text.args = {
  showText: true
}
