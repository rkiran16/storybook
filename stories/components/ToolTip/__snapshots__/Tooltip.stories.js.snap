// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/ToolTip Icon Only 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <script>
      window.addEventListener(\\"DOMContentLoaded\\", () => {
        var tooltipTriggerList = [].slice.call(
          document.querySelectorAll('[data-bs-toggle=\\"tooltip\\"]')
        );
        var tooltipList = tooltipTriggerList.map(function (tooltipTriggerEl) {
          return new bootstrap.Tooltip(tooltipTriggerEl);
        });
      });
    </script>
    <button class=\\"btn btn-link link-secondary text-decoration-none d-flex align-items-center p-2\\" data-bs-toggle=\\"tooltip\\" data-bs-html=\\"true\\" data-bs-placement=\\"bottom\\" title=",
    ">
      ",
    "
      <span class=\\"material-icons-outlined m-0\\">info</span>
    </button>
   ",
  ],
  "values": Array [
    "
    <div class=\\"text-start\\">
      <p class=\\"mb-1\\">
          This is a few lines of tooltip text. 
          This box shows the maximum amount of content that should be displayed inside.
          If more room is needed, use a modal instead.
      </p>   
      <a type=\\"button\\" href=\\"#\\" class=\\"btn btn-tertiary\\">Tertiary Button</a> 
    </div>  
",
    "",
  ],
}
`;

exports[`Storyshots Components/ToolTip With Text 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <script>
      window.addEventListener(\\"DOMContentLoaded\\", () => {
        var tooltipTriggerList = [].slice.call(
          document.querySelectorAll('[data-bs-toggle=\\"tooltip\\"]')
        );
        var tooltipList = tooltipTriggerList.map(function (tooltipTriggerEl) {
          return new bootstrap.Tooltip(tooltipTriggerEl);
        });
      });
    </script>
    <button class=\\"btn btn-link link-secondary text-decoration-none d-flex align-items-center p-2\\" data-bs-toggle=\\"tooltip\\" data-bs-html=\\"true\\" data-bs-placement=\\"bottom\\" title=",
    ">
      ",
    "
      <span class=\\"material-icons-outlined m-0\\">info</span>
    </button>
   ",
  ],
  "values": Array [
    "
    <div class=\\"text-start\\">
      <p class=\\"mb-1\\">
          This is a few lines of tooltip text. 
          This box shows the maximum amount of content that should be displayed inside.
          If more room is needed, use a modal instead.
      </p>   
      <a type=\\"button\\" href=\\"#\\" class=\\"btn btn-tertiary\\">Tertiary Button</a> 
    </div>  
",
    Object {
      "_$litType$": 1,
      "strings": Array [
        "<span class=\\"text-decoration-underline me-2\\">Tooltip Text</span>",
      ],
      "values": Array [],
    },
  ],
}
`;
